internal/strings/strings.go

```diff
@@ -106,3 +106,26 @@ func InSlice(a string, list []string) bool {
 func WrapUnsafe(buf []byte) string {
 	return *(*string)(unsafe.Pointer(&buf))
 }
+
+// AsciiToLower converts the ASCII characters in the string to lowercase.
+// Unlike strings.ToLower, this function does not allocate memory.
+func AsciiToLower(s string) string {
+	for i := 0; i < len(s); i++ {
+		c := s[i]
+		if c >= 'A' && c <= 'Z' {
+			return doAsciiToLower(s, i)
+		}
+	}
+	return s
+}
+
+func doAsciiToLower(s string, start int) string {
+	res := []byte(s)
+	for i := start; i < len(res); i++ {
```

Comment:
since you know that the byte in `start` is a letter, you can do 

```
func doAsciiToLower(s string, start int) string {
	res := []byte(s)
	res[start] += 32
	for i := start + 1; i < len(res); i++ {
		c := res[i]
		if c >= 'A' && c <= 'Z' {
			res[i] += 32
		}
	}
	return WrapUnsafe(res)
}
```